distributed/tests/test_steal.py

```diff
@@ -1869,3 +1855,17 @@ def assert_task_placement(expected, s, workers):
     """Assert that tasks are placed on the workers as expected."""
     actual = _get_task_placement(s, workers)
     assert _equal_placement(actual, expected)
+
+
+# Reproducer from https://github.com/dask/distributed/issues/6573
+@gen_cluster(
```

Comment:
Added reproducer from https://github.com/dask/distributed/issues/6573 for regression testing. This has been solved in #7036.